Backend events

Oporto opened the pull request:
Get and place events through the events servlet

Commit:
Completed base code for part 3 of maps: adding user markers. Version builds and was not tested yet

Commit:
small updates to user-map.html

Commit:
Added favicon

Commit:
Changes to how markers are fetched and display

Commit:
sunday changes

Commit:
Fixed issue with map not loading

Commit:
Removed print statement

Commit:
Added class for Location, ThreadComment, and event classes to data package

Commit:
Updated Events class

Commit:
adding event to Datastore

Commit:
Added store event method

Commit:
Added get and store for events, amenities, and threads to datastore

Commit:
Started Event servlet

Commit:
Finished base get and post Event servlet

Commit:
Finished mvp backend for events

pnvt23 commented on src/main/java/com/google/codeu/data/Location.java:
```diff
@@ -0,0 +1,61 @@
+    private int zipcode;
+    private List<String> adressLines;
+    
+    public Location(double lat, double lng, String name, String country, String state, String city, int zipcode, List<String> adressLines){
```
Can you add some documentation for the constructor?

Usually for this many parameters, it's good idea to use Builder pattern.

pnvt23 commented on src/main/java/com/google/codeu/data/Location.java:
```diff
@@ -0,0 +1,61 @@
+import java.util.List;
+
+public class Location {
+    
```
you should have at most 1 empty line

pnvt23 commented on src/main/java/com/google/codeu/data/Datastore.java:
```diff
@@ -26,71 +26,225 @@
+    *         never received a message. List is sorted by time descending.
+    */
+    public List<Message> getMessages(String recipient) {
+        Query query =
```
Can you fix indentation for this file and the rest ?

Oporto replied:
I fixed indentation and somehow it things that I replaced a big chunk of code, but not edits to the text were made

pnvt23 commented on src/main/java/com/google/codeu/data/Datastore.java:
```diff
@@ -26,71 +26,225 @@
+                Event event = new Event(eventId, timeStamp, speaker, organization, eventDate, location, amenities, externalLink, publicType, ownerId);
+                event.copyThread(thread);
+                events.add(event);
+            } catch (Exception e) {
```
Usually it's good practice to only catch a specific set of Exception, and throw the rest

If there's a need to catch all exception, please add a comment here.

Oporto replied:
Added this line: //An exception can occur here for multiple reasons (Type casting error, any property not existing, key error, etc...)

pnvt23 commented on src/main/java/com/google/codeu/data/Datastore.java:
```diff
@@ -26,71 +26,225 @@
+        }
+        
+        return amenities;
+        
```
Remove empty lines

pnvt23 commented on src/main/java/com/google/codeu/data/Datastore.java:
```diff
@@ -26,71 +26,225 @@
+    public List<ThreadComment> getThread(UUID eventId){
+        List<ThreadComment> thread = new ArrayList<>();
+        
+        Query query = new Query("ThreadComment").setFilter(new Query.FilterPredicate("eventId", FilterOperator.EQUAL, eventId.toString()))
```
can you check the character limit of this line

pnvt23 commented on src/main/java/com/google/codeu/data/Datastore.java:
```diff
@@ -26,71 +26,225 @@
-      }
+    }
+  
+    public List<UserMarker> getMarkers() {
```
I would encourage to document public methods

Oporto replied:
I also updated some methods that should be private

Oporto committed:
Update pom.xml

Commented out configuration

Oporto committed:
Update Datastore.java

Fixed indexing and extra spaces

Oporto committed:
Update Datastore.java

Adjusted public/private methods, updated doc for public methods, and added comments explaining try catch logic

Oporto committed:
Update Datastore.java

Fixed lines with too many characters

Oporto committed:
Update Location.java

Removed extra empty lines and implemented a builder so that a location can be created as:
new Location.Builder()
.setLat(100)
.setLng(200)
....
.build();

Oporto reviewed (commented):
Addressed all comments

Oporto committed:
Merge branch 'master' into backend_events

pnvt23 reviewed (commented):
I still  see the indentation issue for java file. Can you take another look?

It should be +2sp for new block of code, and +4sp if you need to break into multiple lines.

Oporto committed:
Update Datastore.java

Moved indentation to 2sps in Datastore.java

Oporto committed:
Update Event.java

Modified indentation to 2sps in Event.java

Oporto committed:
Update Location.java

Modified indentation to 2sps on Location.java

Oporto committed:
Update UserMarker.java

Modified indentation to 2sps on UserMarker.java

Oporto committed:
Update EventServlet.java

Modified indentation to 2sps on EventServlet.java

Oporto committed:
Update UserMarkerServlet.java

Modified indentation to 2sps on UserMarkerServlet.java

Oporto committed:
Update user-map-loader.js

Modified indentation to 2sps on user-map-loader.js

Oporto committed:
Update user-map.html

Modified indentation on user-map.html to 2 sps

Oporto commented:
I thought it would be 4sp no matter what. Should be fixed now
